web_monitoring/db.py

```diff
@@ -121,6 +122,17 @@ def _build_importable_version(*, page_url, uuid=None, capture_time, uri,
     return version
 
 
+def _validate_timeout(timeout, default=None):
+    if timeout is None:
+        return default
+    elif timeout is not None and timeout < 0:
+        raise ValueError(f'Timeout must be non-negative. "{timeout}" was provided.')
```

Comment:
Since this is an `elif` branch, you don’t need to check `timeout is not None` here.

Reply:
Good catch